### PR TITLE
TooltipContainer - use resize observer instead of getClientBoundingRect

### PR DESCRIPTION
**Pull request by dprokop**
Limits layout calculation for the tooltip container.

before
Screen Shot 2021-03-12 at 09 26 20

after
Screen Shot 2021-03-12 at 10 34 09

### CODE REVIEW

File: packages/grafana-ui/src/components/Chart/TooltipContainer.tsx [hunk 1]
```diff
@@ -20,18 +20,37 @@ export const TooltipContainer: React.FC<TooltipContainerProps> = ({
...
   const { width, height } = useWindowSize();
   const [placement, setPlacement] = useState({
     x: positionX + offsetX,
     y: positionY + offsetY,
   });
 
+  const resizeObserver = new ResizeObserver((entries) => {
+    for (let entry of entries) {
+      const tW = Math.floor(entry.contentRect.width);
+      const tH = Math.floor(entry.contentRect.height);
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/Chart/TooltipContainer.tsx:33]: `contentRect` does not include the 8px padding, maybe add this as fudge here until Safari supports `borderBoxSize`

**Comment by leeoniya** [comment]
LGTM :rocket:

**Review by leeoniya — APPROVED** [review]

File: packages/grafana-ui/src/components/Chart/TooltipContainer.tsx [hunk 2]
```diff
@@ -20,18 +20,37 @@ export const TooltipContainer: React.FC<TooltipContainerProps> = ({
...
+          height: tH,
+        };
+      }
+    }
+  });
+
+  if (tooltipRef.current) {
+    resizeObserver.observe(tooltipRef.current);
+  }
+
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/Chart/TooltipContainer.tsx:47]: i noticed the garbage collector suddenly kicked into overdrive, and then realized all of this is getting called and recreated on every tooltip move. let's make sure this whole block is memoized and only called once.

**Review by leeoniya — CHANGES_REQUESTED** [review]

**Review by hugohaggmark — COMMENTED** [review]
Looks good, but the comments by @leeoniya need to be addressed, I guess?

File: packages/grafana-ui/src/components/Chart/TooltipContainer.tsx (hunk 2 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/Chart/TooltipContainer.tsx:57]: good catch!

**Pull request merged**